### PR TITLE
Increase test coverage for TypeNameSet

### PR DESCRIPTION
**Pull request by khareyash05**
**What this PR does / why we need it**:

The current test coverage for `type_name_set.go` is  77%. [https://app.codecov.io/gh/Azure/azure-service-operator/blob/master/v2/tools/generator/internal/astmodel/type_name_set.go](https://app.codecov.io/gh/Azure/azure-service-operator/blob/master/v2/tools/generator/internal/astmodel/type_name_set.go). I worked on increasing the test coverage by adding more test cases for `Remove()` and `Equals()`

**If applicable**:
- [ ] this PR contains documentation
- [X] this PR contains tests

### CODE REVIEW

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 1]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
 	set.Add(twoTypeName)
 	g.Expect(set.Contains(twoTypeName)).To(BeTrue())
 }
+
+ func Test_TypeNameSet_AfterRemovingOnlyItem_SetIsEmpty(t *testing.T) {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:51]: Add new heading for `Remove() tests`

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 2]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
...
+    t.Parallel()
+    g := NewGomegaWithT(t)
+    set := NewTypeNameSet(oneTypeName)
+    set.Remove(oneTypeName)
+    g.Expect(len(set)).To(Equal(0))
+}
+
+ func Test_TypeNameSet_Equals_ReturnsTrueForEqualSets(t *testing.T) {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:59]: Add new heading for `Equals() Tests`

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 3]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
...
+	t.Parallel()
+	g := NewGomegaWithT(t)
+	set1 := NewTypeNameSet(oneTypeName, twoTypeName)
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:62]: Add a comment highlighting that the two sets are constructed in different orders.

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 4]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
...
+	g.Expect(set1.Equals(set2)).To(BeFalse())
+}
+
+func Test_TypeNameSet_Equals_ReturnsTrueForEmptySets(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+	emptySet1 := NewTypeNameSet()
+	emptySet2 := NewTypeNameSet()
+	g.Expect(emptySet1.Equals(emptySet2)).To(BeTrue())
+}
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:89]: Move into the section for `Empty Set Tests`

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 5]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
...
+
+func Test_TypeNameSet_RemoveNonExistentItem_SetUnchanged(t *testing.T) {
+    t.Parallel()
+    g := NewGomegaWithT(t)
+    set := NewTypeNameSet(oneTypeName)
+    set.Remove(twoTypeName)
+    g.Expect(set.Contains(oneTypeName)).To(BeTrue())
+}
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:97]: Move this and the next test into the section for `Remove() Tests`

File: v2/tools/generator/internal/astmodel/type_name_set_test.go [hunk 6]
```diff
@@ -47,3 +47,68 @@ func Test_TypeNameSet_AfterAddingSecondItem_ContainsItem(t *testing.T) {
...
+    g.Expect(set.Contains(twoTypeName)).To(BeTrue())
+}
+
+func Test_TypeNameSet_RemoveFromEmptySet_SetUnchanged(t *testing.T) {
+    t.Parallel()
+    g := NewGomegaWithT(t)
+    emptySet := NewTypeNameSet()
+    emptySet.Remove(oneTypeName)
+    g.Expect(len(emptySet)).To(Equal(0))
+}
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/type_name_set_test.go:114]: Move into the section for `Empty Set Tests`

**Review by theunrepentantgeek — CHANGES_REQUESTED** [review]
The new test cases look good. 

Tests is this file are in groups (see the existing code comments on lines #20 and #31) so we need to move things around a little. I've added comments on what's needed, and we can merge this once that's done.

Just FYI - our CI is a little flaky at the moment, so you may see us triggering the CI process multiple times. We're working on tracking down the cause.

**Comment by theunrepentantgeek** [comment]
CI failed with the following error:
```
[generator:lint] internal/astmodel/type_name_set_test.go:51: File is not `gofmt`-ed with `-s` (gofmt)
```

I recommend you run our quick checks locally after making changes to catch these kinds of issue. You just needs to run `task` at the root of the repo; see [Contributing ](https://azure.github.io/azure-service-operator/contributing) for more details.

**Comment by theunrepentantgeek** [comment]
CI failed with the following error:

```
[generator:lint] level=info msg="[runner] linters took 12.533784769s with stages: goanalysis_metalinter: 12.530605737s"
[generator:lint] internal/astmodel/type_name_set_test.go:30:1: Function Test_TypeNameSet_RemoveFromEmptySet_SetUnchanged missing the call to method parallel (paralleltest)
[generator:lint] func Test_TypeNameSet_RemoveFromEmptySet_SetUnchanged(t *testing.T) {
[generator:lint] ^
```

Please fix. Recommend running `task generator:lint` locally to catch these before pushing.

I'm looking forward to merging this, really appreciate your efforts.

**Comment by khareyash05** [comment]
@theunrepentantgeek Hi. I am running `task generator:lint` to check if the files are ok to test or not on Github CodeSpaces and I getting this error `task: Command "/workspaces/azure-service-operator/scripts/v2/build_version.py v2" failed: exit status 1`  but when i run `task default` I am getting the list of commands available. I am not able to understand this since its on codespaces and `task default`  is running as well but not `task` or `task generator:lint`

**Comment by khareyash05** [comment]
> 

added the paralleltest

**Comment by khareyash05** [comment]
> CI failed with the following error:
> 
> ```
> [generator:lint] level=info msg="[runner] linters took 12.533784769s with stages: goanalysis_metalinter: 12.530605737s"
> [generator:lint] internal/astmodel/type_name_set_test.go:30:1: Function Test_TypeNameSet_RemoveFromEmptySet_SetUnchanged missing the call to method parallel (paralleltest)
> [generator:lint] func Test_TypeNameSet_RemoveFromEmptySet_SetUnchanged(t *testing.T) {
> [generator:lint] ^
> ```
> 
> Please fix. Recommend running `task generator:lint` locally to catch these before pushing.
> 
> I'm looking forward to merging this, really appreciate your efforts.

Thank you and added the paralleltest

**Review by theunrepentantgeek — APPROVED** [review]

**Comment by theunrepentantgeek** [comment]
Merged! Thanks for the contribution @khareyash05

**Pull request merged**